46.md

```diff
@@ -96,4 +117,10 @@ The signer key will always be the key of the user who controls the signer device
 - **ping**
   - params: []
   - result: `"pong"`
+- **create_account**
+  - params: `["username", "domain", "email"]
```

Comment:
Is email so important to this flow that it's a positional argument? NIP-46 is allergic to options objects, but it would offer a lot more flexibility.

Reply:
it's not really important and can be left empty (all params are optional); we can take it out if it feels too off.

Reply:
Option objects just mean more documentation to read and more strings to write. If we have more options to add later we can add them to the end of the array and leave the email empty.

Reply:
I like this MR. In fact I'm going to implement it. But I need to add a "reason" text field, which the user can fill in to request an account. Requests are approved manually by the admin. The admin account then creates a kind 30361 event to confirm the account of the user: https://gitlab.com/soapbox-pub/ditto/-/blob/main/docs/events.md?ref_type=heads#users 

If approval-mode is disabled, the admin account would create this event manually, and no "reason" field would be required.

Reply:
If this MR goes as it is, I'm going to create a positional 4th argument that is an options object, including `opts.reason `

Reply:
```suggestion
  - params: `["name", "domain", "email"]
```

Username should be `"name"` to match NIP-05.

(`"domain"` should technically be `"host"` so it includes the port, but NIP-05 calls it `"domain"` already)

Also, I wonder if instead it should be a single parameter called `nip05` to match kind 0's metadata field.